src/libraries/System.Net.Security/src/System/Net/Security/Pal.OSX/SafeDeleteNwContext.cs

```diff
@@ -0,0 +1,761 @@
+            try
+            {
+                SafeDeleteNwContext? nwContext = null;
+
+                if (thisHandle != IntPtr.Zero)
+                {
+                    nwContext = ResolveThisHandle(thisHandle);
+                }
+
+                if (NetEventSource.Log.IsEnabled() && statusUpdate != NetworkFrameworkStatusUpdates.DebugLog)
+                    NetEventSource.Info(nwContext, $"Received status update: {statusUpdate}");
+
+                switch (statusUpdate)
+                {
+                    case NetworkFrameworkStatusUpdates.FramerStart:
+                        nwContext?.FramerStartCallback(new SafeNwHandle(Interop.NetworkFramework.Retain(data), true));
+                        break;
+                    case NetworkFrameworkStatusUpdates.FramerStop:
+                        nwContext?.FramerStopCallback();
+                        break;
+                    case NetworkFrameworkStatusUpdates.HandshakeFinished:
+                        nwContext?.HandshakeFinished();
+                        break;
+                    case NetworkFrameworkStatusUpdates.ConnectionFailed:
+                        if (error != null)
+                        {
+                            nwContext?.ConnectionFailed(in *error);
+                        }
+                        break;
+                    case NetworkFrameworkStatusUpdates.ConnectionCancelled:
+                        nwContext?.ConnectionClosed();
+                        break;
+                    case NetworkFrameworkStatusUpdates.CertificateAvailable:
+                        global::System.Runtime.InteropServices.Marshalling.SafeHandleMarshaller<global::Microsoft.Win32.SafeHandles.SafeX509ChainHandle>.ManagedToUnmanagedOut marshaller = new();
+                        marshaller.FromUnmanaged(data);
+                        nwContext?.CertificateAvailable(marshaller.ToManaged());
+                        marshaller.Free();
+                        break;
+                    case NetworkFrameworkStatusUpdates.DebugLog:
+                        HandleDebugLog(nwContext, data, data2);
+                        break;
+                    default: // We shouldn't hit here.
+                        if (NetEventSource.Log.IsEnabled()) NetEventSource.Error(nwContext, $"Unknown status update: {statusUpdate}");
+                        Debug.Assert(false);
+                        break;
+                }
+            }
+            catch (Exception ex)
+            {
+                if (NetEventSource.Log.IsEnabled())
+                {
+                    NetEventSource.Error(null, $"StatusUpdateCallback failed for {statusUpdate}: {ex}");
+                }
+            }
+        }
+        private void FramerStartCallback(SafeNwHandle framerHandle)
+        {
+            _framerHandle = framerHandle;
+        }
+
+        private void FramerStopCallback()
+        {
+            _framerHandle?.Dispose();
+            _framerHandle = null;
+        }
```

Comment:
We dispose the Framer on multiple places (here and in Dispose), we should have only one